vagrant/vagrant-test-sanitizers.sh

```diff
@@ -18,7 +18,23 @@ export ASAN_OPTIONS=strict_string_checks=1:detect_stack_use_after_return=1:check
 export UBSAN_OPTIONS=print_stacktrace=1:print_summary=1:halt_on_error=1
 
 # Run the internal unit tests (make check)
-exectask "ninja-test" "meson test -C build --print-errorlogs --timeout-multiplier=3"
+exectask "ninja-test_sanitizers" "meson test -C build --print-errorlogs --timeout-multiplier=3"
+
+## Run TEST-01-BASIC under test sanitizers
+# Set timeouts for QEMU tests to kill them in case they get stuck
+# As we're not using KVM, bump the QEMU timeout quite a bit
+export QEMU_TIMEOUT=600
+# Set QEMU_SMP to speed things up
+export QEMU_SMP=$(nproc)
+export TEST_NO_NSPAWN=1
+# Arch Linux requires booting with initrd, as all commonly used filesystems
+# are compiled in as modules
+export SKIP_INITRD=no
+
+rm -fr /var/tmp/systemd-test*
+exectask "TEST-01-BASIC_sanitizers" "make -C test/TEST-01-BASIC clean setup run clean-again"
```

Comment:
I think it would be better to run the test in nspawn containers first as we do on Travis CI. Otherwise, we're going to be having a hard time figuring out what can possibly cause a kernel panic as in https://github.com/libbpf/libbpf/pull/29#issuecomment-482340365. In general it's risky to run PID1 under ASan+UBSan in VMs so the idea is to catch as many issues as possible in containers first. In that particular case, the test would have failed run via `nspawn` because on Arch the hybrid hierarchy is used in containers so we wouldn't have had to deal with that kernel panic at all.

Reply:
I actually thought it's the other way around - running potentially unstable PID 1 under nspawn, where it communicates with host's PID 1, sounded more dangerous to me. But with the kernel panic stuff it actually makes sense to run the nspawn test beforehand.